utils/process/src/lib.rs

```diff
@@ -14,6 +14,25 @@ use {
     shared_child::unix::SharedChildExt,
 };
 
+#[cfg(unix)]
+pub trait CommandSetSid<T> {
+    fn setsid(&mut self) -> &mut T;
+}
+
+#[cfg(unix)]
+impl CommandSetSid<Command> for Command {
+    fn setsid(&mut self) -> &mut Command {
+        use std::os::unix::process::CommandExt;
+        unsafe {
+            self.pre_exec(|| {
+                libc::setsid();
```

Comment:
Is it intentional we don't check for errors? Also, perhaps it might be better to use a wrapper [`nix::setsid()`](https://docs.rs/nix/0.18.0/nix/unistd/fn.setsid.html)?

Reply:
> Is it intentional we don't check for errors?

Kind of - we still want to continue execution (this shouldn't fail btw). I'll return an `Err` and add a debug statement in case of failure, even `warn` seems a bit too much.

>  Also, perhaps it might be better to use a wrapper `nix::setsid()`?

At first I've implemented it with `nix` in the provider agent, but eventually decided to move it to this crate without adding extra dependencies. I'll revert to `nix` anyway.